capsules/core/src/spi_controller.rs

```diff
@@ -165,17 +169,19 @@ impl<'a, S: SpiMasterDevice<'a>> Spi<'a, S> {
                         .unwrap_or(0),
                 });
             app.index += read_len;
-            self.spi_master.read_write_bytes(
-                self.kernel_write.take().unwrap(),
-                self.kernel_read.take(),
-                read_len,
-            )
+            let kwbuf = self
+                .kernel_write
+                .take()
+                .unwrap_or((&mut [] as &'static mut [u8]).into());
+            self.spi_master
+                .read_write_bytes(kwbuf, self.kernel_read.take())
```

Comment:
Same with `read_len` here.

Userspace might use a single large buffer, but want to do smaller SPI operations at times. It seems like that might be broken now